quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java

```diff
@@ -26,4 +26,24 @@ public UserEntity getUserById(final String userId) {
             return null;
         }
     }
+
+    /**
+     * Delete a user by given id from the DB.
+     *
+     * @param userId Id of the user whose information is to be fetched.
+     * @return User details which is to be deleted if exist in the DB else null.
+     */
+
+    public UserEntity deleteUser(final String userId){
+        try {
+            UserEntity deleteUser = this.entityManager.createNamedQuery( "userByUserId", UserEntity.class ).
```

Comment:
Use the existing code. 
UserEntity deleteUser = getUserById(userId);
Null check required. 
1. The getUserById function may result in NRE and return null.
2. In case of race condition getUserById is expected to return null.
if(deleteUser!=null){
this.entityManager.remove( deleteUser );
}

Reply:
This is already taken care in try-catch block and is going to return null for NRE.

Reply:
Updated the code with reuseable code.